airflow/models/dag.py

```diff
@@ -450,219 +454,167 @@ def date_range(
         num: Optional[int] = None,
         end_date: Optional[datetime] = timezone.utcnow(),
     ) -> List[datetime]:
+        message = "`DAG.date_range()` is deprecated."
         if num is not None:
-            end_date = None
-        return utils_date_range(
-            start_date=start_date, end_date=end_date, num=num, delta=self.normalized_schedule_interval
-        )
+            result = utils_date_range(start_date=start_date, num=num)
+        else:
+            message += " Please use `DAG.get_run_dates(..., align=False)` instead."
+            result = self.get_run_dates(start_date, end_date, align=False)
+        warnings.warn(message, category=DeprecationWarning, stacklevel=2)
+        return result
 
     def is_fixed_time_schedule(self):
-        """
-        Figures out if the DAG schedule has a fixed time (e.g. 3 AM).
-
-        :return: True if the schedule has a fixed time, False if not.
-        """
-        now = datetime.now()
-        cron = croniter(self.normalized_schedule_interval, now)
-
-        start = cron.get_next(datetime)
-        cron_next = cron.get_next(datetime)
-
-        if cron_next.minute == start.minute and cron_next.hour == start.hour:
+        warnings.warn(
+            "`DAG.is_fixed_time_schedule()` is deprecated.",
+            category=DeprecationWarning,
+            stacklevel=2,
+        )
+        try:
+            return not self.timetable._schedule._should_fix_dst
+        except AttributeError:
             return True
 
-        return False
-
     def following_schedule(self, dttm):
         """
         Calculates the following schedule for this dag in UTC.
 
         :param dttm: utc datetime
         :return: utc datetime
         """
-        if isinstance(self.normalized_schedule_interval, str):
-            # we don't want to rely on the transitions created by
-            # croniter as they are not always correct
-            dttm = pendulum.instance(dttm)
-            naive = timezone.make_naive(dttm, self.timezone)
-            cron = croniter(self.normalized_schedule_interval, naive)
-
-            # We assume that DST transitions happen on the minute/hour
-            if not self.is_fixed_time_schedule():
-                # relative offset (eg. every 5 minutes)
-                delta = cron.get_next(datetime) - naive
-                following = dttm.in_timezone(self.timezone) + delta
-            else:
-                # absolute (e.g. 3 AM)
-                naive = cron.get_next(datetime)
-                tz = pendulum.timezone(self.timezone.name)
-                following = timezone.make_aware(naive, tz)
-            return timezone.convert_to_utc(following)
-        elif self.normalized_schedule_interval is not None:
-            return timezone.convert_to_utc(dttm + self.normalized_schedule_interval)
+        current = pendulum.instance(dttm)
+        between = TimeRestriction(None, None, catchup=True)
```

Comment:
```suggestion
        between = TimeRestriction(earliest=None, latest=None, catchup=True)
```

easier to understand :) -- again not strictly needed